assets/aml-benchmark/components/src/inference_postprocessor/inference_postprocessor.py

```diff
@@ -90,129 +134,606 @@ def __post_init__(self) -> None:
+        return data
+
+    def apply_remove_prompt_prefix(self, text: str, data: dict = None) -> str:
+        """Remove prompts that has been added as prefix in the given text."""
+        prompt_prefix = get_prompt(data, self.remove_prompt_prefix)
+        if prompt_prefix and text.startswith(prompt_prefix):
+            text = text[len(prompt_prefix):]
+        return text
+
+    def apply_separator(self, text: str):
+        """Apply few shot separator used in prompt crafter."""
+        if self.separator:
+            # self.separator = json.loads(json.dumps(self.separator))
+            self.separator = self._convert_to_unicode(self.separator)
+            text = text.split(self.separator)[0]
+        return text
+
+    def run_processor_using_template(self, key: str = None) -> None:
         """Postprocessor run using template."""
         result_df = pd.DataFrame()
-        if self.ground_truth_dataset:
-            actual_df = pd.json_normalize(read_jsonl_files(resolve_io_path(self.ground_truth_dataset)))
-        if self.y:
-            result_df[self.y] = actual_df[self.y]
-        else:
-            result_df = actual_df
+        result_df = self.read_ground_truth_dataset(result_df)
+        # read the prediction dataset
         predicted_data = read_jsonl_files(resolve_io_path(self.prediction_dataset))
         pred_list = []
-        if self.Y in predicted_data[0].keys():
-            key = self.Y
+        if self.prediction_column_name in predicted_data[0].keys():
+            key = self.prediction_column_name
         else:
-            key = "0"
+            key = key if key else "0"
         template = self.template
         env = jinja2.Environment()
         jinja_template = env.from_string(template)
         for row in predicted_data:
-            if key == "0":
-                row[self.Y] = row.get(key)
-            out_string = jinja_template.render(row)
-            pred_list.append(out_string)
-        result_df[self.Y] = pred_list
-        if self.pred_probs_dataset:
-            probs_df = pd.json_normalize(read_jsonl_files(resolve_io_path(self.pred_probs_dataset)))
-            if self.encoder_config:
-                self.encoder_config = json.loads(self.encoder_config)
-                probs_df = probs_df.rename(columns=self.encoder_config).add_prefix('probs_')
-                result_df = pd.concat([result_df, probs_df], axis=1)
-        result_df.to_json(self.result, lines=True, orient='records')
+            if key != self.prediction_column_name:
+                row[self.prediction_column_name] = row.get(key)
+            predicted = row.get(self.prediction_column_name)
+            if isinstance(predicted, list):
+                try:
+                    out_string = jinja_template.render(predicted)
+                    pred_list.append(out_string)
+                except Exception as e:
+                    # "jinja2.exceptions.UndefinedError: 'list object' has no attribute 'split'"

```

Comment:
When is this exception thrown? What is the scenario? Also, can you update the code comment to provide a brief context for it as well? Thanks!